ans-4175/talenta-api
# Feat/scheduler

# Pull Request Template

## Description

In this PR, I add scheduler features. I use `node-cron`, it is nodejs scheduler library. Then I also add `pm2` to run process in the background. 

I made several change in index.js and execute.js. I also add scripts in package.json to run the scheduler
- scheduler:start, to start scheduler for clockin and for clockout
- scheduler:stop, to delete pm2 process to stop scheduler

For now, user need to define the times in config.js file, 
```
{
  timeClockIn: "hh:mm",
  timeClockOut: "hh:mm",
}
```

Fixes #8 

## Type of change

Please delete options that are not relevant.


## How Has This Been Tested?

How to reproduce test:
- you may comment the line 38 and 40 in index.js, to prevent it from hit the real API. Just to make sure that the scheduler is running on time. You may look at commit ab5af1e. 
- Define your `timeClockIn` and `timeClockOut`
- Open two terminal, one for monitoring, run `yarn pm2 monit`
- In the second terminal, run `yarn scheduler:start`
- Wait for the log to appear at time you've defined. 

This is screenshot how I am tested this. (Not hit real API)
<img width="1440" alt="Screen Shot 2021-10-05 at 20 50 45" src="https://user-images.githubusercontent.com/39824404/136042448-f0e7106d-f405-44a8-b1be-3cb8e86c0444.png">


## Checklist:
